src/analysis.ts

```diff
@@ -0,0 +1,284 @@
+  prepareExtendingBundle,
+  determineBaseDir,
+  resolveBundleFilePath,
+} from './files';
+import parseGitUri from './gitUtils';
+import {
+  getFilters,
+  createGitBundle,
+  GetAnalysisErrorCodes,
+  getAnalysis,
+  AnalysisStatus,
+  IResult,
+  GetAnalysisResponseDto,
+  AnalysisFailedResponse,
+  AnalysisFinishedResponse,
+} from './http';
+import emitter from './emitter';
+import { defaultBaseURL, MAX_PAYLOAD, IGNORES_DEFAULT } from './constants';
+import { prepareRemoteBundle, fullfillRemoteBundle } from './bundles';
+
+import { AnalysisSeverity, IGitBundle, IFileBundle, IBundleResult } from './interfaces/analysis-result.interface';
+
+async function pollAnalysis(
+  baseURL: string,
+  sessionToken: string,
+  bundleId: string,
+  useLinters: boolean,
+  severity: AnalysisSeverity,
+): Promise<IResult<AnalysisFailedResponse | AnalysisFinishedResponse, GetAnalysisErrorCodes>> {
+  let analysisResponse: IResult<GetAnalysisResponseDto, GetAnalysisErrorCodes>;
+  let analysisData: GetAnalysisResponseDto;
+
+  emitter.analyseProgress({
+    status: AnalysisStatus.fetching,
+    progress: 0,
+  });
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    // eslint-disable-next-line no-await-in-loop
+    analysisResponse = await getAnalysis({
+      baseURL,
+      sessionToken,
+      bundleId,
+      useLinters,
+      severity,
+    });
+
+    if (analysisResponse.type === 'error') {
+      return analysisResponse;
+    }
+
+    analysisData = analysisResponse.value;
+
+    if (
+      analysisData.status === AnalysisStatus.fetching ||
+      analysisData.status === AnalysisStatus.analyzing ||
+      analysisData.status === AnalysisStatus.dcDone
+    ) {
+      // Report progress of fetching
+      emitter.analyseProgress(analysisData);
+    } else if (analysisData.status === AnalysisStatus.done) {
+      // Return data of analysis
+      return analysisResponse as IResult<AnalysisFinishedResponse, GetAnalysisErrorCodes>;
+      // deepcode ignore DuplicateIfBody: false positive it seems that interface is not taken into account
+    } else if (analysisData.status === AnalysisStatus.failed) {
+      // Report failure of analysing
+      return analysisResponse as IResult<AnalysisFailedResponse, GetAnalysisErrorCodes>;
+    }
+  }
+}
+
+export async function analyzeBundle(
+  baseURL = defaultBaseURL,
+  sessionToken = '',
+  includeLint = false,
+  severity = AnalysisSeverity.info,
+  bundleId: string,
+  baseDir = '',
+): Promise<IBundleResult> {
+  // Call remote bundle for analysis results and emit intermediate progress
+  const analysisData = await pollAnalysis(baseURL, sessionToken, bundleId, includeLint, severity);
+
+  if (analysisData.type === 'error') {
+    throw analysisData.error;
+  } else if (analysisData.value.status === AnalysisStatus.failed) {
+    throw new Error('Analysis has failed');
+  }
+
+  const { analysisResults } = analysisData.value;
+
+  const filesPositions = Object.fromEntries(
```

Comment:
This logic should not be in analyzeBundle and should be moved to analyzeFoldes as analyzeBundle is called by analyzeGit and should not have any knowledge of system directories

Reply:
done